### PR TITLE
Prune unnecessary information from TransportNodesStatsAction.NodeStatsRequest 

### PR DESCRIPTION
**Pull request by NEUpanning**
Today every node-level TransportNodesStatsAction.NodeStatsRequest carries the entire top-level NodesStatsRequest, but it only needs to carry the metrics. The uneccessary information like nodes list incurs extra effort and network traffic.This commit trims these things down.

Relates #100878

### CODE REVIEW

**Review by kingherc — COMMENTED** [review]
Thanks for the effort! Some comments.

File: server/src/main/java/org/elasticsearch/TransportVersions.java [hunk 1]
```diff
@@ -177,6 +177,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion TRANSFORM_GET_CHECKPOINT_QUERY_AND_CLUSTER_ADDED = def(8_544_00_0);
     public static final TransportVersion GRANT_API_KEY_CLIENT_AUTHENTICATION_ADDED = def(8_545_00_0);
     public static final TransportVersion PIT_WITH_INDEX_FILTER = def(8_546_00_0);
+    public static final TransportVersion NODE_STATS_REQUEST_SIMPLIFIED = def(8_547_00_0);
```
[the remark this answers is not in the document]
Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/TransportVersions.java]: The confict resolved in efd5a9817fe9e6afd2859be19145cce4ec311bfc

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java [hunk 2]
```diff
@@ -51,6 +51,11 @@ public NodesStatsRequest(String... nodesIds) {
         nodesStatsRequestParameters = new NodesStatsRequestParameters();
     }
 
+    public NodesStatsRequest(NodesStatsRequestParameters nodesStatsRequestParameters, String... nodesIds) {
+        super(nodesIds);
```
[the remark this answers is not in the document]
Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java:54]: Sure, it is refactored in a1fcc60e9f5bdd5eb41e992ddea31a6c33dc888c

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java [hunk 3]
```diff
@@ -104,32 +108,54 @@ protected NodeStats nodeOperation(NodeStatsRequest nodeStatsRequest, Task task)
...
 
         @Override
         public Task createTask(long id, String type, String action, TaskId parentTaskId, Map<String, String> headers) {
             return new CancellableTask(id, type, action, "", parentTaskId, headers) {
                 @Override
                 public String getDescription() {
-                    return request.getDescription();
+                    return Strings.format(
+                        "nodes=%s, metrics=%s, flags=%s",
+                        Arrays.toString(nodesIds),
```
[the remark this answers is not in the document]
Comment (NEUpanning) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/TransportNodesStatsAction.java:138]: I agree with you and the nodesIds variable is not as expensive as the `DiscoveryNode` list.

File: docs/changelog/102559.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+pr: 102559
+summary: "Prune unnecessary information from TransportNodesStatsAction.NodeStatsRequest"
+area: Stats
```
[the remark this answers is not in the document]
Comment (NEUpanning) [reply docs/changelog/102559.yaml]: Sure, it's done in 8cbb01212fc4c4d2ba5e857f99e35dee8877dfff.

**Comment by kingherc** [comment]
buildkite test this

**Comment by kingherc** [comment]
One test failure I saw seems to have been there already - opened https://github.com/elastic/elasticsearch/issues/103119 . Will check tests again before approving.

**Review by kingherc — APPROVED** [review]
LGTM. Thanks for the PR! Merging it.

**Pull request merged**